doc/upgrade/6.10.md

```diff
@@ -0,0 +1,166 @@
+# Upgrade steps from 6.9 to 6.10
+
+## Field Type external storage
+
+### External storage gateways
+
+Abstract base class `eZ\Publish\Core\FieldType\StorageGateway` for the gateway of a Field Type external storage and the
+`eZ\Publish\Core\FieldType\StorageGateway::setConnection` method are deprecated.
+
+Extend the `eZ\Publish\SPI\FieldType\StorageGateway` class and inject Database Handler directly
```

Comment:
to better signal that the change is not something field types have to do immediately, start the sentence by saying something like:

`To prepare your FieldType for future storage engines and avoid using the deprecated abstract, extend the ..`